Introduce Bolt 5.7 and 5.8 (#68)
neo4j/docs-bolt: modules/ROOT/pages/bolt/message.adoc

*** The scheme `basic` requires a username `principal::String` and a password `credentials::String`.
*** The scheme `bearer` merely requires a token `credentials::String`.

The following fields are defined for inclusion in the `SUCCESS` metadata.

> ** `advertised_address::String?` - the configured advertised address of the database server. 

Author: I will not be present if you're using unix sockets, for example. The advertised_address can be misleading on that situation.

label:new[Introduced in bolt 5.8]

> ** `credentials_expired::Boolean` - the sent credentials are expired, the user must update it.

Reviewer: ```suggestion
** `credentials_expired::Boolean` - the sent credentials are expired, the user must update them.
```

> ** `credentials_expired::Boolean` - the sent credentials are expired, the user must update it.

Author: I think this information since ever on the LOGON method response. I just realised it when I was implementing advertised_address

> ** `credentials_expired::Boolean` - the sent credentials are expired, the user must update it.

Reviewer 2: > And shouldn't it at least be marked optional?

I guess that's not right. It's not that it's optional but that it defaults to `false`, I assume?

> ** `credentials_expired::Boolean` - the sent credentials are expired, the user must update it.

Author: This is false by default

LOGON {"scheme": "basic", "principal": "user", "credentials": "password"}
----

[[messages-logon-success]]
==== Server response `SUCCESS`

A `SUCCESS` message response indicates that the client has been successfully authenticated.

The following fields are defined for inclusion in the `SUCCESS` metadata.

** `advertised_address::String?` - the configured advertised address of the database server.
label:new[Introduced in bolt 5.8]

> ** `credentials_expired::Boolean` - the sent credentials are expired, the user must update them.

Author: ```suggestion
** `credentials_expired::Boolean?` - the sent credentials are expired, the user must update them. Default: `false`.
```

> ** `credentials_expired::Boolean?` - the sent credentials are expired, the user must update them. Default: `false`.

Author: ```suggestion
** `credentials_expired::Boolean` - the sent credentials are expired, the user must update them. If this field is not present, it should be considered to be `false`.
```

[source, bolt]
----
C: TELEMETRY 2
S: FAILURE {"neo4j_code":"Neo.ClientError.Request.Invalid","message":"Message of type TelemetryMessage cannot be handled by a session in the NEGOTIATION state.","gql_status":"50N42","description":"error: general processing exception - unexpected error. Unexpected error has occurred. See debug log for details."}
----

> NOTE: This error still as unexpected (`50N42`) since it didn't get migrate yet to GQL standard.

Reviewer 2: This requires a whole lot of background knowledge in how we handle the GQL error migration. I'm in favor of not going into this here. Either make up a status code or don't comment on it, I'd say. It's just an example and no guarantee that the server does actually send these exact values.

** `fields::List<String>`, the fields of the return result. e.g. [“name”, “age”, …]
** `t_first::Integer`, the time, specified in ms, which the first record in the result stream is available after.

For *Implicit Transaction* (`RUN`):

> ** `db::String?` - user's resolved home database that the transaction will run into. 

Reviewer: ```suggestion
** `db::String?` - resolved user's home database that the transaction will run into. 
```

[[messages-begin-success]]
==== Server response `SUCCESS`

The following fields are defined for inclusion in the `SUCCESS` metadata:

> ** `db::String?` - user's resolved home database that the transaction will run into. 

Reviewer: ```suggestion
** `db::String?` - resolved user's home database that the transaction will run into. 
```

.Fields:
[source, bolt]
----
metadata::Dictionary
----

The following fields are defined in the metadata:

* `message::String` - the human readable description of the failure.
* `code::String` - the neo4j code identifying the failure.
label:deprecated[Removed after bolt 5.6]
* `neo4j_code::String` - the neo4j code identifying the failure.
label:new[Introduced in bolt 5.7]

> * `gql_status::String` - the gql status which identifies the error. This should be used in favor of `neo4j_code`.

Reviewer: ```suggestion
* `gql_status::String` - the GQL status which identifies the error. This should be used in favor of `neo4j_code`.
```

See more about GQL Status Code https://development.neo4j.dev/docs/status-codes/current/[here].
label:new[Introduced in bolt 5.7]
* `description::String` - describes the failure represented by `gql_status`.
label:new[Introduced in bolt 5.7]

> * `diagnostic_record::Dictionary` - contains fields for helping diagnostic the status. Omitted when content is default.

Reviewer: ```suggestion
* `diagnostic_record::Dictionary` - contains fields for helping diagnosing the status. Omitted when content is default.
```
